nautobot_secrets_providers/providers/delinea.py

```diff
@@ -176,21 +177,33 @@ def query_thycotic_secret_server(  # pylint: disable=too-many-boolean-expression
             else:
                 delinea = SecretServer(base_url=base_url, authorizer=thy_authorizer)
 
-            # Attempt to retrieve the secret.
-            try:
-                if secret_id is not None:
-                    secret = ServerSecret(**delinea.get_secret(secret_id))
-                else:
-                    secret = ServerSecret(**delinea.get_secret_by_path(secret_path))
-            except SecretServerError as err:
-                raise exceptions.SecretValueNotFoundError(secret, caller_class, str(err)) from err
+            ###
+            # Current issue: https://github.com/DelineaXPM/python-tss-sdk/issues/25
+            # below for loop is an attempt at a workaround
+            ###
+            for retry in range(4):  # try 4 times and then give up.
```

Comment:
Beyond my comment about trying to solve this in nautobot-plugin-nornir, I've tended to like `backoff` https://pypi.org/project/backoff/